src/zenml/integrations/whylogs/data_validators/whylogs_data_validator.py

```diff
@@ -0,0 +1,176 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the whylogs data validator."""
+
+import datetime
+from typing import Any, ClassVar, Optional, Sequence, cast
+
+import pandas as pd
+import whylogs as why  # type: ignore
+from whylogs.api.writer.whylabs import WhyLabsWriter  # type: ignore
+from whylogs.core import DatasetProfileView  # type: ignore
+
+from zenml.data_validators import BaseDataValidator
+from zenml.environment import Environment
+from zenml.integrations.whylogs import WHYLOGS_DATA_VALIDATOR_FLAVOR
+from zenml.integrations.whylogs.secret_schemas.whylabs_secret_schema import (
+    WhylabsSecretSchema,
+)
+from zenml.logger import get_logger
+from zenml.repository import Repository
+from zenml.steps import STEP_ENVIRONMENT_NAME, StepEnvironment
+
+logger = get_logger(__name__)
+
+
+class WhylogsDataValidator(BaseDataValidator):
+    """Whylogs data validator stack component.
+
+    Attributes:
+        whylabs_secret: Optional ZenML secret with Whylabs credentials. If
+            configured, all the data profiles returned by all pipeline steps
+            will automatically be uploaded to Whylabs in addition to being
+            stored in the ZenML Artifact Store.
+    """
+
+    whylabs_secret: Optional[str] = None
```

Comment:
I really like how you did this. I did something similar for the annotation component, just with less style.

Reply:
Hopefully this is converted to an even more elegant style with a refactor of the syntax of expressing secrets in the stack component

Reply:
Are you telling me to update how I did my version, or is this comment directed at @stefannica?

Reply:
It was directed at me, in light of this ongoing discussion: https://github.com/zenml-io/zenml/issues/752

Reply:
@strickvl check @schustmi 's comment too, there's a mixin class that makes this even easier and reduces code duplication.

Reply:
Ah yes. Possible the mixin class could be made a bit more generic / abstract? The language now for that feels like it's just for authenticating with cloud providers.